plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -306,18 +314,7 @@ private static ConnectorPageSource createOrcPageSource(
                 }
 
                 if (orcColumn != null) {
-                    Type readType;
-                    if (column.getType() == UUID) {
-                        if (!"UUID".equals(orcColumn.getAttributes().get(ICEBERG_BINARY_TYPE))) {
```

Comment:
@findepi how important was this check? I took it out to avoid needing to traverse the `orcColumn` in tandem with the column type, but if it's important I can add it back in.

Reply:
Per iceberg spec, the column should have the attribute set, so if it is not set, something is wrong.
how hard would it be to have this check?

Reply:
I tried writing it back in but it winds up being a lot of work. Effectively needing to traverse the `OrcColumn` tree along with the `Type` tree, while dealing with the Iceberg Field IDs potentially not existing. It's not pretty.

Reply:
Let's add a TODO that we should have validation. If you have code that works, we should keep the validation. We can figure out how to improve the code later.

Reply:
Added a TODO, I didn't manage to get something working but I can give it another shot when I have time.